tinybmp/src/lib.rs

```diff
@@ -208,18 +205,23 @@ impl<'a> Iterator for BmpIterator<'a> {
                 self.x = 0;
             }
 
-            // Shorter alias for easier to read code
-            let start = self.start_idx;
+            let byte_idx = self.bit_idx / 8;
 
             let pixel_value = match self.pixel_stride {
-                1 => u32::from(px[start]),
-                2 => u32::from_le_bytes([px[start], px[start + 1], 0, 0]),
-                3 => u32::from_le_bytes([px[start], px[start + 1], px[start + 2], 0]),
-                4 => u32::from_le_bytes([px[start], px[start + 1], px[start + 2], px[start + 3]]),
+                1 => (px[byte_idx] & 0b_1000_0000 >> self.bit_idx % 8 > 0) as u32,
```

Comment:
IMO this is a little hard to read. Can you change this into a block to make it more readable? Something like this:
```suggestion
                1 => {
                    let mask = 0b_1000_0000 >> self.bit_idx % 8;
                    (px[byte_idx] & mask != 0) as u32
                }
```

Reply:
Thanks that is definitely better to read. I will add that in.